CI-faq-q2 (#7494)
harness/developer-hub: kb/continuous-integration/continuous-integration-faqs.md

### How can user prevent conflicts with Save Cache steps in parallel stages?
Skip the **Save Cache** step in all parallel stages except one.
Use conditional execution in the **Save Cache** step to ensure it runs only in one instance of the parallel stages.

### What conditional execution user add to the Save Cache step to handle parallel stages?
```
when:
  stageStatus: Success
  condition: <+strategy.iteration> == 0

```
This ensures the **Save Cache** step runs only if it's the first instance (iteration 0) of the parallel stages and if the current stage execution is successful.

### How does caching work when cloning pipelines with Save/Restore Cache steps?
When the user clones a pipeline with caching steps, the cache keys generated by the cloned pipeline use the original pipeline's cache key as a prefix. For instance, if the original cache key is `some-cache-key`, the cloned pipeline might use `some-cache-key2`. This prevents accidental interference between cache keys of the original and cloned pipelines.

### How can user ensure correct cache handling when cloning pipelines with Save Cache to GCS?

> Enable separators (/) for GCS cache keys by setting PLUGIN_ENABLE_SEPARATOR: true in your pipeline's stage variables.

Reviewer: ```suggestion
Enable separators (/) for GCS cache keys by setting `PLUGIN_ENABLE_SEPARATOR: true` in your pipeline's stage variables.
```

### How can user resolve S3 bucket permission issue in save cache in S3 step?

> If aws connector is configured with the Cross account role ARN, so user need to configure the PLUGIN_USER_ROLE_ARN stage variable as suggested in the below doc:
> https://developer.harness.io/docs/continuous-integration/use-ci/build-and-upload-artifacts/build-and-push/build-and-push-to-ecr-step-settings/#aws-connector

Reviewer: ```suggestion
If the AWS connector is configured with the cross-account role ARN, the user needs to configure the `PLUGIN_USER_ROLE_ARN` stage variable as suggested in [this doc](../../docs/continuous-integration/use-ci/build-and-upload-artifacts/build-and-push/build-and-push-to-ecr-step-settings/#aws-connector)
```

### Can user import test results from a generic 'mvn clean deploy' Run step into the test Dashboard? Or must use the Unit Test step to display them on that tab?

> Yes, you can use the Run step to run your maven command, and upload the test results. There is a field for this in the run step (like in the run tests step). Please see
> https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/containerized-steps/run-step/#report-paths for more information.

Reviewer: ```suggestion
Yes, you can use the Run step to run your maven command and upload the test results. There is a field for this in the run step (like in the run tests step). Please check [the documentation](../../docs/continuous-delivery/x-platform-cd-features/cd-steps/containerized-steps/run-step/#report-paths) for more information.
```
